### PR TITLE
PocketKit - Complete Concurrency #1

### PR DESCRIPTION
**Pull request by Gio2018**
## Summary
* This PR enables complete concurrency on `PocketKit` and addresses some warnings

## References 
* usual

## Implementation Details
* See title and code

## Test Steps
* Test everything

## PR Checklist:
- [x] Added Unit / UI tests - More like updated them
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

### CODE REVIEW

File: PocketKit/Sources/Analytics/Tracker/PocketSnowplowTracker.swift [hunk 1]
```diff
@@ -2,13 +2,13 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-import SnowplowTracker
+@preconcurrency import SnowplowTracker
```
Comment (bassrock) [inline PocketKit/Sources/Analytics/Tracker/PocketSnowplowTracker.swift:5]: We should open an issue on https://github.com/snowplow/snowplow-ios-tracker/issues

File: PocketKit/Sources/PocketKit/LoggedOut/LoggedOutViewModel.swift [hunk 2]
```diff
@@ -166,8 +163,9 @@ class LoggedOutViewModel: ObservableObject {
 
     private func authenticate(_ authentication: (ASWebAuthenticationPresentationContextProviding?) async throws -> AuthorizationClient.Response) async {
         do {
+            // TODO: CONCURRENCY - Need to figure out how to handle ASWebAuthenticationPresentationContextProviding and other native non-sendable types
```
Comment (bassrock) [inline PocketKit/Sources/PocketKit/LoggedOut/LoggedOutViewModel.swift:164]: figures apple would require something and their own classes arent sendable.

File: PocketKit/Sources/PocketKit/SceneTracker.swift [hunk 3]
```diff
@@ -7,7 +7,8 @@ import Analytics
 import SwiftUI
 import Combine
 
-class SceneTracker {
+@MainActor
```
Comment (bassrock) [inline PocketKit/Sources/PocketKit/SceneTracker.swift:10]: Does this need MainActor, I don't see anything that needs the main thread?

**Review by bassrock — APPROVED** [review]

File: PocketKit/Sources/PocketKit/SceneTracker.swift (hunk 3 above)
Comment (Gio2018) [reply PocketKit/Sources/PocketKit/SceneTracker.swift:10]: in that class, we listen for two notifications, that generate the following warnings:
`Main actor-isolated class property 'didActivateNotification' can not be referenced from a non-isolated context; this is an error in Swift 6`

`Main actor-isolated class property 'didEnterBackgroundNotification' can not be referenced from a non-isolated context; this is an error in Swift 6`

if we add `@MainActor` to the specific methods, we then have to also add it to `init`, which basically means adding the type itself

File: PocketKit/Sources/Analytics/Tracker/PocketSnowplowTracker.swift (hunk 1 above)
Comment (Gio2018) [reply PocketKit/Sources/Analytics/Tracker/PocketSnowplowTracker.swift:5]: done. I figure this will not be the only case...

**Pull request merged**